src/content-tooltip/components/container.jsx

```diff
@@ -27,10 +36,89 @@ class TooltipContainer extends React.Component {
         showTooltip: false,
         position: { x: 250, y: 200 },
         tooltipState: 'copied',
+        highlightsOn: false,
     }
 
-    componentDidMount() {
+    async componentDidMount() {
         this.props.onInit(this.showTooltip)
+
+        const {
+            shortcutsEnabled,
+            highlightShortcut,
+            linkShortcut,
+            toggleSidebarShortcut,
+            toggleHighlightsShortcut,
+            createAnnotationShortcut,
+            highlightShortcutEnabled,
+            linkShortcutEnabled,
+            toggleSidebarShortcutEnabled,
+            toggleHighlightsShortcutEnabled,
+            createAnnotationShortcutEnabled,
+        } = await getKeyboardShortcutsState()
+
+        const handleKeyboardShortcuts = e => {
```

Comment:
I was confused at first at your reply in the other thread when you said "otherwise I will have to clutter up the state with all the settings variables", but now I see what you mean. But there's still a fairly simple way to move this `handleKeyboardShortcuts` event handler out of `componentDidMount`: make it a higher order function, with the outer-most function taking in the keyboard shortcut state.

eg:
```js
initHandleKeyboardShortcuts = state => e => {
  // ... all your conditional logic
}

async componentDidMount() {
  const state = await getKeyboardShortcutsState()

  if (state.shortcutsEnabled) {
    Mousetrap.bind([
      // ...
    ], this.initHandleKeyboardShortcuts(state))
  }
}
```